kafka-rest/src/main/java/io/confluent/kafkarest/GroupMetadataObserver.java

```diff
@@ -0,0 +1,307 @@
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.Properties;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.HashSet;
+import java.util.Comparator;
+
+
+public class GroupMetadataObserver {
+
+  private static KafkaConsumer<?, ?> createConsumer(String groupId,
+                                                              KafkaRestConfig appConfig) {
+    final Properties properties = new Properties();
+    String deserializer = StringDeserializer.class.getName();
+    properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+            RestConfigUtils.bootstrapBrokers(appConfig));
+    properties.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+    properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+    properties.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "30000");
+    properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, deserializer);
+    properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, deserializer);
+    properties.putAll(appConfig.getConsumerProperties());
+
+    return new KafkaConsumer<>(properties);
+  }
+
+  private final Logger log = LoggerFactory.getLogger(GroupMetadataObserver.class);
+
+  private final KafkaRestConfig config;
+  private final AdminClientWrapper adminClientWrapper;
+
+  public GroupMetadataObserver(KafkaRestConfig config, AdminClientWrapper adminClientWrapper) {
+    this.config = Objects.requireNonNull(config);
+    this.adminClientWrapper = Objects.requireNonNull(adminClientWrapper);
+  }
+
+  /**
+   * <p>Get consumer group list restricted by paging parameters</p>
+   *
+   * @return list of consumer groups
+   */
+  public List<ConsumerGroup> getPagedConsumerGroupList(Integer startPos,
+                                                       Integer count)
+      throws Exception {
+    return getConsumerGroups(getPagedConsumerGroup(startPos, count));
+  }
+
+  /**
+   * <p>Get consumer group list</p>
+   *
+   * @return list of consumer groups
+   */
+  public List<ConsumerGroup> getConsumerGroupList() throws Exception {
+    return getConsumerGroups(adminClientWrapper.listConsumerGroups());
+  }
+
+  private List<ConsumerGroup> getConsumerGroups(Collection<ConsumerGroupListing> groupsOverview)
+      throws Exception {
+    final List<ConsumerGroup> result = new ArrayList<>();
+    for (ConsumerGroupListing eachGroupInfo : groupsOverview) {
+      final Node node = adminClientWrapper.describeConsumerGroups(
```

Comment:
Do we need to describe each group in a separate call? Why not call it once for all groups?